File: solutions/typescript/resistor-color-duo/2/resistor-color-duo.ts
export const COLORS: string[] = [
      'black',
      'brown',
      'red',
      'orange',
      'yellow',
      'green',
      'blue',
      'violet',
      'grey',
      'white',
    ];

export function decodedValue(colors: string[]): number {
  let first: number = COLORS.indexOf(colors[0])
  let second: number = COLORS.indexOf(colors[1])
  
  return (first * 10) + second 
}

File: solutions/ruby/resistor-color-duo/3/resistor_color_duo.rb
class ResistorColorDuo
  COLOR_CODE = {
        "black": 0,
        "brown": 1,
        "red": 2,
        "orange": 3,
        "yellow": 4,
        "green": 5,
        "blue": 6,
        "violet": 7,
        "grey": 8,
        "white": 9
    }
  
  def self.value(colors)
    tens = COLOR_CODE[colors[0].to_sym] * 10
    ones = COLOR_CODE[colors[1].to_sym]
    
    tens + ones
  end
end